platform/markdown/src/main/java/br/com/orcinus/orca/platform/markdown/annotated/AnnotatedString.extensions.kt

```diff
@@ -0,0 +1,222 @@
+import android.text.ParcelableSpan
+import androidx.compose.runtime.State
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.ShaderBrush
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.graphics.drawscope.DrawStyle
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import br.com.orcinus.orca.platform.markdown.span.toSpanStyle
+import kotlin.reflect.full.primaryConstructor
+
+/**
+ * Converts this [AnnotatedString] into a [State] that holds an [Editable].
+ *
+ * @param context [Context] with which conversions from [ParcelableSpan]s into [SpanStyle]s are
+ *   performed.
+ * @throws IllegalArgumentException If any of the [SpanStyle]s specifies a [Brush] isn't a
+ *   [SolidColor] nor a [ShaderBrush], since there aren't equivalent [ParcelableSpan]s for [Brush]es
+ *   other than those of such types.
+ * @throws NoSuchMethodException If the specified [Brush] is a [ShaderBrush] or a [DrawStyle] has
+ *   been defined and the primary constructor of `androidx.compose.ui:ui-text`'s `DrawStyleSpan` or
+ *   `ShaderBrushSpan` isn't found when converting [SpanStyle]s into [ParcelableSpan]s, given that
+ *   they're referenced and called through reflection because both are APIs are internal to the
+ *   module in which they've been declared as of 1.6.6.
+ * @see ParcelableSpan.toSpanStyle
+ * @see AnnotatedString.spanStyles
+ * @see SpanStyle.brush
+ * @see SpanStyle.drawStyle
+ * @see primaryConstructor
+ */
+@Throws(IllegalArgumentException::class, NoSuchMethodException::class)
+internal fun AnnotatedString.toEditableAsState(context: Context): State<Editable> {
+  return buildState {
+    object : Editable, CharSequence by text {
+      /**
+       * [InputFilter]s that have been set for constraining changes made to this [Editable].
+       *
+       * @see setFilters
+       * @see getFilters
+       */
+      private var filters = emptyArray<InputFilter>()
+
+      /**
+       * Alias for `getSpans(0, length(), ParcelableSpan.class)`, which obtains all of the
+       * [ParcelableSpan]s that have been set to this [Editable] (which is the only type of span
+       * that it can contain).
+       *
+       * @see getSpans
+       */
+      private val spans
+        get() = getSpans(start = 0, end = length, ParcelableSpan::class.java)
+
+      override fun getChars(start: Int, end: Int, dest: CharArray?, destoff: Int) {
+        dest?.let {
+          var index = destoff
+          for (character in slice(start..end.dec())) {
+            it[index++] = character
+          }
+        }
+      }
+
+      override fun <T : Any?> getSpans(start: Int, end: Int, type: Class<T>?): Array<T> {
+        val spanCount = spanStyles.size
+        val coercedStart = start.coerceIn(0, spanCount)
+        val coercedEnd = end.coerceIn(0, spanCount)
+
+        @Suppress("UNCHECKED_CAST")
+        return if (coercedStart == 0 && coercedEnd == 0) {
```

Comment:
An empty array should be returned when both the start and the end index are equal, not only when they’re zero.